alpha/lib/model/categoryEntry.php

```diff
@@ -219,5 +219,230 @@ public function setCreatorPuserId ($v)
 	{
 		$this->putInCustomData('creatorPuserId', $v);
 	}
+
+	/**
+	 * @param $entryId
+	 * @param $categoryId
+	 * @throws KalturaErrors::CATEGORY_NOT_FOUND
+	 * @throws KalturaErrors::MAX_CATEGORIES_FOR_ENTRY
+	 * @throws KalturaErrors::CANNOT_ASSIGN_ENTRY_TO_CATEGORY
+	 * @throws KalturaErrors::CATEGORY_ENTRY_ALREADY_EXISTS*
+	 * @throws KalturaErrors::INVALID_ENTRY_ID
+	 * @throws Exception
+	 * @return categoryEntry $categoryEntry
+	 */
+	public function add($entryId, $categoryId)
+	{
+		$entry = $this->retrieveEntry($entryId);
+		$category = $this->retrieveCategory($categoryId);
+
+		$this->validateMaxCategoriesPerEntry($entry, $category);
+		$currentKsKuserId = kCurrentContext::getCurrentKsKuserId();
+		$this->validateKuserEntitledToAssignEntryToCategory($category, $entry, $currentKsKuserId);
+		$categoryEntryExists = categoryEntryPeer::retrieveByCategoryIdAndEntryId($categoryId, $entryId);
+		$this->validateCategoryEntryNotExist($categoryEntryExists);
+
+		if($categoryEntryExists)
+		{
+			$categoryEntryExists->copyInto($this);
+		}
+
+		$this->setStatus(CategoryEntryStatus::ACTIVE);
+		$this->handleModeration($category, $currentKsKuserId);
+		$this->assignPartnerId();
+		$this->assignCreator();
+	}
 	
+	/**
+	 * Set CategoryEntry as deleted and check entitelement
+	 * @throws KalturaErrors::INVALID_ENTRY_ID
+	 * @throws KalturaErrors::CATEGORY_NOT_FOUND
+	 * @throws KalturaErrors::CANNOT_REMOVE_ENTRY_FROM_CATEGORY
+	 */
+	public function setAsDeleted()
+	{
+		$entryId = $this->entry_id;
+		$categoryId = $this->category_id;
+		
+		$entry = entryPeer::retrieveByPK($entryId);
+		
+		if (!$entry)
+		{
+			if (kCurrentContext::$master_partner_id != Partner::BATCH_PARTNER_ID)
+				throw new KalturaAPIException(KalturaErrors::INVALID_ENTRY_ID, $entryId);
+		}
+		
+		$category = categoryPeer::retrieveByPK($categoryId);
+		if (!$category && kCurrentContext::$master_partner_id != Partner::BATCH_PARTNER_ID)
+			throw new KalturaAPIException(KalturaErrors::CATEGORY_NOT_FOUND, $categoryId);
+		
+		//validate user is entitled to remove entry from category
```

Comment:
add function validateKuserEntitledToRemoveEntryToCategory that checks the entitlments